src/ralph/backends/data/swift.py

```diff
@@ -330,10 +334,9 @@ def write(  # noqa: PLR0912, PLR0913
                 msg = "Failed to write to object %s: %s"
                 error = err.msg
                 logger.error(msg, target_object, error)
-                if not ignore_errors:
-                    raise BackendException(msg % (target_object, error)) from err
+                raise BackendException(msg % (target_object, error)) from err
 
-        count = sum(1 for _ in data)
+        count = counter["count"]
```

Comment:
My understanding is that `counter` will reflect the number times `data` generator is read in the above line (`self.connection.put_object(...)`), in the `try` block. But if there is an error during "PUT", the value of `count` will be off by 1 (since the data was already read).

More hypothetic, if reading is done by batch through some internal logic and there is an error, then the value of count could be off by an unknown value.

Opened this issue for testing: https://github.com/openfun/ralph/issues/522

Reply:
Nice catch) Thanks) Yes, you are right; we indeed should increment the counter after yielding the statement (if we start to count with 0). Good idea to open an issue for testing this further. (Updated the counter)